File: core/src/main/java/com/thefuntasty/taste/display/TDisplay.java
package com.thefuntasty.taste.display;

import android.content.Context;
import android.util.DisplayMetrics;
import android.view.WindowManager;

import com.thefuntasty.taste.Taste;

public class TDisplay {

	private final Context context;

	public TDisplay(Context context) {
		this.context = context;
	}

	public int getWindowWidth() {
		DisplayMetrics metrics = new DisplayMetrics();
		WindowManager windowManager = (WindowManager) Taste.context().getSystemService(Context.WINDOW_SERVICE);
		windowManager.getDefaultDisplay().getMetrics(metrics);
		return metrics.widthPixels;
	}

	public int getWindowHeight() {
		DisplayMetrics metrics = new DisplayMetrics();
		WindowManager windowManager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
		windowManager.getDefaultDisplay().getMetrics(metrics);
		return metrics.heightPixels;
	}
}

File: core/src/main/java/com/thefuntasty/taste/display/TDisplay.kt
package com.thefuntasty.taste.display

import android.content.Context
import android.util.DisplayMetrics
import android.view.WindowManager

class TDisplay(private val context: Context) {

	val windowWidth: Int
		get() {
			val metrics = DisplayMetrics()
			val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
			windowManager.defaultDisplay.getMetrics(metrics)
			return metrics.widthPixels
		}

	val windowHeight: Int
		get() {
			val metrics = DisplayMetrics()
			val windowManager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
			windowManager.defaultDisplay.getMetrics(metrics)
			return metrics.heightPixels
		}

	fun orientation(): Int  = context.resources.configuration.orientation

	fun dpToPx(dp: Float): Float = dp * context.resources.displayMetrics.density

	fun pxToDp(px: Float): Float = px / context.resources.displayMetrics.density
}
